### PR TITLE
Extend to all image sizes > 128

### PR DESCRIPTION
**Pull request by AlessioGalluccio**
If the sizes of the image are not divisible by 128, the upscaled tensors could not match the correct dimensions. This fix uses interpolation and more precise upsampling to fix this problem

### CODE REVIEW

**Comment by marco-rudolph** [comment]
While I appreciate that you are contributing to this repository, I do not want to include it for the following reasons: 

- It is a special case that can be fixed by changing the input size
- The proposed technique is not part of the original paper and has not been evaluated. This repo is mainly about reproducing the paper. This should happen in a minimal way if possible, so that I have to supervise/moderate as little as possible here, as I don't want to provide a permanent service.
- The syntax is partly not standard ("if(x)" instead of "if x")

Basically I am open to minor changes. Anything beyond that is not the idea of the repository.

I really don't mean any harm and I'm also happy that people are interested in the work and want to contribute actively. I hope you can understand that.

**Comment by AlessioGalluccio** [comment]
I didn't know that if(x) wasn't standard syntax. Thank you for the feedback! I coded in C and Java in the past. I can understand your idea, no harm in that. This repo must be first and foremost loyal to the paper. 
I'm using this repo in my master thesis, so contributing to this project is a way to say thanks. I will open pull requests only for minor changes in the future then.

I know that this is not the right place, but can I ask you a thing about cs-flow that I can't understand? Why 768x768 resize? I ask this because this net seems to get worse results if I use a different size, even in datasets that have images of rectangular shapes (1024x256 in my case). This happened even before I used this fix.

**Comment by marco-rudolph** [comment]
Ah okay, I already guessed...
Thank you and good luck with your thesis! :)

Why 768? That is really a good question. Tbh I just experimented with a few image sizes and this led to the best results. But it is really surprising that this holds in your case.

**Comment by AlessioGalluccio** [comment]
So, it just happened to be the best size. This is quite curious.
Thank you very much!
